### ISSUE TITLE
Task parameter inputs/arguments separation

### ISSUE THREAD

**Issue by elliotgunton** [argoproj-labs/hera-workflows#433]
Figured this out while exploring related to #430 and investigating how to pass dag inputs to the tasks underneath.

As we define Tasks and form the dag from them, the dag's tasks.arguments are implicitly built from all the `Task` inputs - we cannot separate the `value` we want to provide to both. E.g. this spec is impossible to define in Hera as far as I can tell:

```yaml
spec:
  entrypoint: my-dag
  templates:
  - container:
      args:
      - print('got inputs', {{inputs.parameters}})
      command:
      - python
      - -c
      image: python:3.9.10
    inputs:
      parameters:
      - name: my-param  # Note there is no value provided here ...
    name: my-task
  - dag:
      tasks:
      - arguments:
          parameters:
          - name: my-param
            value: '{{inputs.parameters.my-param}}'  # ...while I can provide a value here (which uses the dag's "my-param")
        name: my-task
        template: my-task
    inputs:
      parameters:
      - name: my-param
    name: my-dag
```

This spec would enforce me to provide an input for `my-param` whether I wanted to launch the dag or `my-task` directly.

To compare, in Hera, I can define this (which works) but the Task input takes the value `{{inputs.parameters.my-param}}`  which becomes an empty string unless I change it:

```py
    with WorkflowTemplate(name="parameters-from-dag-to-task", dag_name="my-dag") as wt:
        t = Task(
            name="my-task",
            command=["python", "-c"],
            inputs=[Parameter(name="my-param", value="{{inputs.parameters.my-param}}")],
            outputs=[Parameter(name="output-param")],
            args=["print('got inputs', {{inputs.parameters}})"],
        )

        wt.dag.inputs = [Parameter(name="my-param")]

    return wt
```

The spec I get from this is:

```yaml
spec:
  entrypoint: my-dag
  templates:
  - container:
      args:
      - print('got inputs', {{inputs.parameters}})
      command:
      - python
      - -c
      image: python:3.9.10
    inputs:
      parameters:
      - name: my-param
        value: '{{inputs.parameters.my-param}}'  # this value is actually pointing to itself - not very obvious what happens (empty string)
    name: my-task
    outputs:
      parameters:
      - name: output-param
  - dag:
      tasks:
      - arguments:
          parameters:
          - name: my-param
            value: '{{inputs.parameters.my-param}}'  # this is correct
        name: my-task
        template: my-task
    inputs:
      parameters:
      - name: my-param # this is correct
    name: my-dag
```

In a nutshell: should we expose the `dag.tasks.arguments` separately from the Task template's `inputs`?

**Comment by elliotgunton** [comment]
templates and tasks are being split out in V5 work, fixed in https://github.com/argoproj-labs/hera-workflows/pull/451
